k8s/eks/bash/functions.sh

```diff
@@ -0,0 +1,312 @@
+    availabilityZones: ["$AVAILABILITY_ZONE"]
+    # ssh:
+    #   allow: true
+    #   # key needs to exist
+    #   publicKeyPath: $SSH_PATH_INFRA
+    iam:
+      attachPolicyARNs:
+        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryFullAccess
+        - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
+        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+        - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
+   
+    overrideBootstrapCommand: |
+      #!/bin/bash
+      /etc/eks/bootstrap.sh $CLUSTER_NAME \
+        --kubelet-extra-args '--max-pods=58' \
+        --use-max-pods false
+  - name: ng-2-plan
+    labels:
+      "testground.node.role.plan": "true"
+    instanceType: $INSTANCE_TYPE_PLAN 
+    # Amazon EKS optimized Amazon Linux 2 v1.22 built on 08 Aug 2022
+    ami: $AMI_PLAN
+    desiredCapacity: $DESIRED_CAPACITY_PLAN
+    volumeSize: $VOLUME_SIZE_PLAN
+    privateNetworking: false
+    availabilityZones: [$AVAILABILITY_ZONE]
+    # ssh:
+    #   allow: true
+    #   publicKeyPath: $SSH_PATH_PLAN
+    iam:
+      attachPolicyARNs:
+        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryFullAccess
+        - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
+        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+        - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
+    preBootstrapCommands:
+      - "echo -e 'fs.file-max=3178504\nnet.core.somaxconn = 131072\nnet.netfilter.nf_conntrack_max = 1048576\nnet.core.netdev_max_backlog = 524288\nnet.core.rmem_max = 16777216\nnet.core.wmem_max = 16777216\nnet.ipv4.tcp_rmem = 16384 131072 16777216\nnet.ipv4.tcp_wmem = 16384 131072 16777216\nnet.ipv4.tcp_mem = 262144 524288 1572864\nnet.ipv4.tcp_max_syn_backlog = 131072\nnet.ipv4.ip_local_port_range = 10000 65535\nnet.ipv4.tcp_tw_reuse = 1\nnet.ipv4.ip_forward = 1\nnet.ipv4.conf.all.rp_filter = 0\nnet.ipv4.neigh.default.gc_thresh2 = 4096\nnet.ipv4.neigh.default.gc_thresh3 = 32768' >> /tmp/999-testground.conf; sudo cp /tmp/999-testground.conf /etc/sysctl.d/999-testground.conf; sudo sysctl -p /etc/sysctl.d/999-testground.conf; echo -e '*  soft  nproc  131072\n*  hard  nproc  262144\n*  soft  nofile 131072\n*  hard  nofile 262144' >> /tmp/999-limits.conf; sudo cp /tmp/999-limits.conf /etc/security/limits.d/999-limits.conf"
+    # 234 is the max number of pods for c5.4xlarge
+    overrideBootstrapCommand: |
+      #!/bin/bash
+      /etc/eks/bootstrap.sh $CLUSTER_NAME \
+       --kubelet-extra-args '--max-pods=234' \
+       --use-max-pods false
+EOT
+}
+
+jq_check(){ 
+has_jq_installed=$(which jq)
+if [[ "$OSTYPE" == "linux-gnu"* ]]
+ then
+   if [ -z $has_jq_installed ]
+     then
+       sudo apt update
+       sudo apt install jq
+   else
+       echo ""
+   fi
+   else
+       echo "Unsupported operating system. This will work for 'linux-gnu'." 
+ fi
+}
```

Comment:
This should be listed in the requirements instead.

Reply:
Agree; currently it only works for linux hosts. We will be adding changes to make it work on macOS as well.

Reply:
Don't, add it to the list of requirements.

There is a lot against doing this, one obvious reason is that we don't need that feature and we don't want to maintain it.

Reply:
Ok, understood. Will be added as a TODO.

Reply:
I don't think installing `jq` as a side effect of `testground_install` is a good idea. I'd rather see it clearly listed as a script requirement - left for the user to decide how to install it.

Reply:
We will look into the best way of achieving this, thanks for the suggestion.

Reply:
`jq` has been listed as a requirement, and no longer installed by the script.